src/StoreApi/Schemas/V1/CartItemSchema.php

```diff
@@ -458,8 +458,19 @@ protected function get_item_data( $cart_item ) {
 		 * @param array $cart_item Cart item array.
 		 * @return array
 		 */
-		$item_data = apply_filters( 'woocommerce_get_item_data', array(), $cart_item );
-		return array_map( [ $this, 'format_item_data_element' ], $item_data );
+		$item_data           = apply_filters( 'woocommerce_get_item_data', array(), $cart_item );
+		$formatted_item_data = array_map( [ $this, 'format_item_data_element' ], $item_data );
+
+		// Remove empty arrays from the data.
+		$valid_item_data = array_filter(
+			$formatted_item_data,
+			function ( $item ) {
+				return ! empty( $item );
```

Comment:
Since this would also filter out empty strings and `0` and other falsey things, should we filter out null instead, and tweak `format_item_data_element` to convert non-scalar values to `null`?